ta/secure_key_services/src/processing_ec.c

```diff
@@ -1083,7 +1083,14 @@ uint32_t sks2tee_ecdh_param_pub(struct sks_attribute_head *proc_params,
 
 	*pub_size = temp;
 
-	return serialargs_get_ptr(&args, pub_data, temp);
+	/* Skip DER tag */
+	rv = serialargs_get(&args, &temp, sizeof(uint8_t));
+	if (rv || temp != 0x04)
+		return rv;
+
+	*pub_size -= sizeof(uint8_t);
```

Comment:
The format of the public key is in DER-encoding of ANSI X9.62 right? If so we should check if this is DER long or short and do the respective conversion. This change works for der short, but won't work for der long, so please at least add a TODO here so we can later add der long (which is also needs to be done at sks2tee_load_attr).

Reply:
Maybe mbedtls can help for DER conversion (yet i've not checked :disappointed:)

Reply:
um I am not sure about any of this. do you have an example? this is just addressing the way an encoded octet string is being parsed (ie, we must to skip the 04 tag)
are you saying that for DER long, octect strings are encoded differently?

Reply:
@ricardosalveti thanks! yes things are as you pointed out  (below). will add an EMSG on returning !0x04 until we support 0x02 or 0x03

ECC public keys have the following syntax:
     ECPoint ::= OCTET STRING

   Implementations of Elliptic Curve Cryptography according to this
   document MUST support the uncompressed form and MAY support the
   compressed form of the ECC public key.  The hybrid form of the ECC
   public key from [X9.62] MUST NOT be used.  As specified in [SEC1]:

      o The elliptic curve public key (a value of type ECPoint that is
        an OCTET STRING) is mapped to a subjectPublicKey (a value of
        type BIT STRING) as follows: the most significant bit of the
        OCTET STRING value becomes the most significant bit of the BIT
        STRING value, and so on; the least significant bit of the OCTET
        STRING becomes the least significant bit of the BIT STRING.
        Conversion routines are found in Sections 2.3.1 and 2.3.2 of
        [SEC1].

      o The first octet of the OCTET STRING indicates whether the key is
        compressed or uncompressed.  The uncompressed form is indicated
        by 0x04 and the compressed form is indicated by either 0x02 or
        0x03 (see 2.3.3 in [SEC1]).  The public key MUST be rejected if
        any other value is included in the first octet.